galera/src/replicator_str.cpp

```diff
@@ -994,6 +1011,17 @@ ReplicatorSMM::request_state_transfer (void* recv_ctx,
     // in the sst_received.
     sst_state_ = SST_WAIT;
 
+    // We need to release the mutex before sending the request. This is
```

Comment:
Yes. It gets stuck inside `send_state_request`, because of the same reason I explained above. 

Receiver thread will wait indefinitely for an update on its queue. This is usually updated by the group communication thread. But, since the Gcomm thread cannot connect to the donor, it never updates the queue.